src/npc/behaviour/chicken_behaviour_tree.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import random
+from dataclasses import dataclass
+
+import pygame
+
+from src.npc.bases.chicken_base import ChickenBase
+from src.npc.behaviour.ai_behaviour_tree_base import Context, Selector, Action
+from src.settings import SCALED_TILE_SIZE
+
+
+@dataclass
+class ChickenBehaviourTreeContext(Context):
+    chicken: ChickenBase
+
+
+class ChickenBehaviourTree:
+    """
+    Group of classes used for Chicken behaviour.
```

Comment:
All instances of "Group of **classes**" should be replaced by "group of **methods**", since it’s functions you use, not classes

Reply:
You're right, I'll change that right away 😅